pkg/util/deployment/deployment_test.go

```diff
@@ -75,3 +79,256 @@ func TestGetReadyPodsCount(t *testing.T) {
+			Name:   api.SimpleNameGenerator.GenerateName("rc"),
+			Labels: labels,
+		},
+		Spec: api.ReplicationControllerSpec{
+			Replicas: 1,
+			Selector: labels,
+			Template: &api.PodTemplateSpec{
+				Spec: api.PodSpec{
+					Containers: []api.Container{
+						{
+							Name:                   image,
+							Image:                  image,
+							ImagePullPolicy:        api.PullAlways,
+							TerminationMessagePath: api.TerminationMessagePathDefault,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+// generateRC creates a replication controller, with the input deployment's template as its template
+func generateRC(deployment extensions.Deployment) api.ReplicationController {
+	template := GetNewRCTemplate(deployment)
+	return api.ReplicationController{
+		ObjectMeta: api.ObjectMeta{
+			Name:   api.SimpleNameGenerator.GenerateName("rc"),
+			Labels: template.Labels,
+		},
+		Spec: api.ReplicationControllerSpec{
+			Template: &template,
+			Selector: template.Labels,
+		},
+	}
+}
+
+// generateDeployment creates a deployment, with the input image as its template
+func generateDeployment(image string) extensions.Deployment {
+	podLabels := map[string]string{"name": image}
+	terminationSec := int64(30)
+	return extensions.Deployment{
+		ObjectMeta: api.ObjectMeta{
+			Name: image,
+		},
+		Spec: extensions.DeploymentSpec{
+			Replicas:       1,
+			Selector:       podLabels,
+			UniqueLabelKey: "deployment.kubernetes.io/podTemplateHash",
+			Template: api.PodTemplateSpec{
+				ObjectMeta: api.ObjectMeta{
+					Labels: podLabels,
+				},
+				Spec: api.PodSpec{
+					Containers: []api.Container{
+						{
+							Name:                   image,
+							Image:                  image,
+							ImagePullPolicy:        api.PullAlways,
+							TerminationMessagePath: api.TerminationMessagePathDefault,
+						},
+					},
+					DNSPolicy:                     api.DNSClusterFirst,
+					TerminationGracePeriodSeconds: &terminationSec,
+					RestartPolicy:                 api.RestartPolicyAlways,
+					SecurityContext:               &api.PodSecurityContext{},
+				},
+			},
+		},
+	}
+}
+
+func TestGetNewRC(t *testing.T) {
+	newDeployment := generateDeployment("nginx")
+	newRC := generateRC(newDeployment)
+
+	tests := []struct {
+		test     string
+		rcList   api.ReplicationControllerList
+		expected *api.ReplicationController
+	}{
+		{
+			"No new RC",
+			api.ReplicationControllerList{
+				Items: []api.ReplicationController{
+					generateRC(generateDeployment("foo")),
+					generateRC(generateDeployment("bar")),
+				},
+			},
+			nil,
+		},
+		{
+			"Has new RC",
+			api.ReplicationControllerList{
+				Items: []api.ReplicationController{
+					generateRC(generateDeployment("foo")),
+					generateRC(generateDeployment("bar")),
+					generateRC(generateDeployment("abc")),
+					newRC,
+					generateRC(generateDeployment("xyz")),
+				},
+			},
+			&newRC,
+		},
+	}
+
+	ns := api.NamespaceDefault
+	for _, test := range tests {
+		c := &simple.Client{
```

Comment:
I can only set one request in simple.Client, but I need GET both replicationcontrollers and pods for GetOldRCs, so I changed the function to only test its main logic.